app/models/user.rb

```diff
@@ -24,4 +24,16 @@ class User < ApplicationRecord
   def to_s
     email
   end
+
+  def add_role(role)
+    role = Role.find_by(name: role)
+    role = Role.create(name: role) if role.nil?
```

Comment:
I think `Role.create` works like a find_or_create, but this is clearer.

Reply:
I added that because it wasn't doing find_or_create and was raising an error